lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -943,4 +943,78 @@ public void detach_success_callback_detached() throws AblyException {
 				ably.close();
 		}
 	}
+
+	/**
+	 * <p>
+	 * Validate publish will result in an error, when the channel is
+	 * in or moves to the FAILED state before the operation succeeds
+	 * </p>
+	 * <p>
+	 * Spec: RTL6c3
+	 * </p>
+	 *
+	 * @throws AblyException
+	 */
+	@Test
+	public void attach_implicit_publish_fail() throws AblyException {
+		AblyRealtime ably = null;
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = testVars.createOptions(testVars.keys[1].keyStr);
+			ably = new AblyRealtime(opts);
+
+			/* wait until connected */
+			new ConnectionWaiter(ably.connection).waitFor(ConnectionState.connected);
+			assertEquals("Verify failed state reached", ably.connection.state, ConnectionState.connected);
+
+			/* create a channel and subscribe */
+			final Channel channel = ably.channels.get("publish_fail");
+			channel.publish("Lorem", "Ipsum!");
+			assertEquals("Verify attaching state reached", channel.state, ChannelState.attaching);
+
+			ErrorInfo fail = new ChannelWaiter(channel).waitFor(ChannelState.failed);
+			assertEquals("Verify failed state reached", channel.state, ChannelState.failed);
+			assertEquals("Verify reason code gives correct failure reason", fail.statusCode, 401);
+		} finally {
+			if(ably != null)
+				ably.close();
+		}
+	}
+
+	/**
+	 * <p>
+	 * Validate subscribe will result in an error, when the channel is
+	 * in or moves to the FAILED state before the operation succeeds
+	 * </p>
+	 * <p>
+	 * Spec: RTL7c
+	 * </p>
+	 *
+	 * @throws AblyException
+	 */
+	@Test
```

Comment:
This test suffers from the same problem as the `publish()` test above: it does not test what it says, and all of the assertions will simply be bypassed by the exception thrown by `subscribe()`.